### PR TITLE
Validate html

### PR DESCRIPTION
**Pull request by ryanto**
WIP - This PR will allow `visit` to expose a boolean based on if fastboot returned a valid html document.

### CODE REVIEW

File: addon-test-support/-private/html-parser.js [hunk 1]
```diff
@@ -0,0 +1,252 @@
+/*
```
Comment (stefanpenner) [inline addon-test-support/-private/html-parser.js]: does it make sense to inline an html-parser, or use one that is actively maintained from NPM?

Comment (stefanpenner) [reply addon-test-support/-private/html-parser.js]: http://npmjs.com/package/parse5 maybe?

Comment (ryanto) [reply addon-test-support/-private/html-parser.js]: I'd love to find another solution here. I feel like using this inline html parser is going to be very brittle and a headache!

The problem I'm running into is how to parse the HTML w/o doing auto correction. That way I can compare a parsed tree to `DOMParser`/`parse5`, and see if all the nodes match up. If they don't match up, we know there's going to be rehydration problems.

I guess if I could snap my fingers, the ideal thing would be having something like `parse5` tell me if it did any correction? Do you know of any like that?

As an example, how do I know parse5 corrected my `p > div` here: https://astexplorer.net/#/gist/ba58b92ec191fb97aa0d83e502517c1b/0156a79828eceec42613517789a53f2385bc6cf7

**Comment by ryanto** [comment]
Gonna close this, the more I think about this PR it's really about rehydration working correctly. Happy to come back it in once we land rehydration.

**Pull request closed without merging**